airbyte_cdk/sources/declarative/retrievers/file_uploader.py

```diff
@@ -0,0 +1,44 @@
+import json
+from pathlib import Path
+from typing import Optional
+
+from airbyte_cdk.sources.declarative.extractors.record_extractor import RecordExtractor
+from airbyte_cdk.sources.declarative.partition_routers.substream_partition_router import (
+    SafeResponse,
+)
+from airbyte_cdk.sources.declarative.requesters import Requester
+from airbyte_cdk.sources.declarative.types import Record, StreamSlice
+
+
+class FileUploader:
+    def __init__(
+        self,
+        requester: Requester,
+        download_target_extractor: RecordExtractor,
+        content_extractor: Optional[RecordExtractor] = None,
+    ) -> None:
+        self._requester = requester
+        self._download_target_extractor = download_target_extractor
+        self._content_extractor = content_extractor
+
+    def upload(self, record: Record) -> None:
+        # TODO validate record shape - is the transformation applied at this point?
```

Comment:
There were a couple of discussions that makes me think this is needed:
* Adding a AirbyteFileRecordMessage to the protocol instead of relying on a implicit class that we only have ¾in the CDK [here](https://github.com/airbytehq/airbyte-python-cdk/blob/c4d0f918170dc48d8af95d21244d3f796ba9e3d1/airbyte_cdk/models/file_transfer_record_message.py#L8-L13)
* Supporting canonical schemas

In order words, this TODO logic here would validate that we can create the AirbyteMessage that describes a file

Reply:
This has been outscoped for now and it seems like the record can have any shape now. I would probably add the attachment ID so that users can rely on a foreign key and the would be it